src/recipe/recipe.controller.spec.ts

```diff
@@ -1,18 +1,175 @@
+import { PrismaService } from 'src/prisma/prisma.service';
 
-describe('RecipeController', () => {
-  let controller: RecipeController;
+describe("Recipe Controller's tests", () => {
+    let recipeController: RecipeController;
+    let recipeService: RecipeService;
+    let prismaService: PrismaService;
 
-  beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      controllers: [RecipeController],
-    }).compile();
+    const mockRecipe = {
+        id: expect.any(Number),
+        createdAt: expect.any(Number),
+        title: expect.any(String),
+        description: expect.any(String),
+        ingredients: expect.any(String),
+        preparation: expect.any(String),
+    };
 
-    controller = module.get<RecipeController>(RecipeController);
-  });
+    class mockRecipeRequest {
+        title: string;
+        description: string;
+        ingredients: string;
+        preparation: string;
+    }
 
-  it('should be defined', () => {
-    expect(controller).toBeDefined();
-  });
+    const mockRecipeService = {
+        createRecipe: jest.fn((dto: mockRecipeRequest) => {
+            return {
+                id: 1,
+                createdAt: Date.now(),
+                ...dto,
+            };
+        }),
+        fetchRecipe: jest.fn((id: number) => {
+            return {
+                id,
+                createdAt: Date.now(),
+                title: 'Fetched recipe title',
+                description: 'Fetched recipe description',
+                ingredients: 'Fetched recipe ingredients',
+                preparation: 'Fetched recipe preparation',
+            };
+        }),
+        fetchAllRecipes: jest.fn(() => {
+            return [
+                {
+                    id: 0,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+                {
+                    id: 1,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+            ];
+        }),
+        updateRecipe: jest.fn((id: number, dto: mockRecipeRequest) => {
+            return {
+                id,
+                createdAt: Date.now(),
+                title: dto.title || 'latest title',
+                description: dto.description || 'latest description',
+                ingredients: dto.ingredients || 'latest ingredients',
+                preparation: dto.preparation || 'latest preparation',
+            };
+        }),
+        deleteRecipe: jest.fn((_id: number) => {
```

Comment:
why `_id` instead of `id`?

Reply:
I wanted to mark it as 'unused value' but i couldn't find any good convention on internet for this. So i sticked with Rust one tbh

Reply:
ah I see now - yes adding underscore at the beginning is in fact they correct way to mark the variable as unused in the TypeScript 👍🏻

ok, then let's do it this way: check out the comment below - if you decide to go with `deleteRecipe: jest.fn()` then problem solved 👍🏻 If you decide with other option (with implementation) go ahead and use `_id` 👍🏻